src/tools/perf/perftest.c

```diff
@@ -476,16 +499,21 @@ static ucs_status_t parse_test_params(ucx_perf_params_t *params, char opt, const
         }
         return UCS_OK;
     case 'D':
+        optarg2 = strchr(optarg, ',');
         if (0 == strcmp(optarg, "short")) {
             params->uct.data_layout   = UCT_PERF_DATA_LAYOUT_SHORT;
         } else if (0 == strcmp(optarg, "bcopy")) {
             params->uct.data_layout   = UCT_PERF_DATA_LAYOUT_BCOPY;
         } else if (0 == strcmp(optarg, "zcopy")) {
             params->uct.data_layout   = UCT_PERF_DATA_LAYOUT_ZCOPY;
-        } else if (0 == strcmp(optarg, "iov")) {
-            params->ucp.datatype      = UCP_PERF_DATATYPE_IOV;
-        } else if (0 == strcmp(optarg, "contig")) {
-            params->ucp.datatype      = UCP_PERF_DATATYPE_CONTIG;
+        } else if (UCS_OK == parse_ucp_datatype_params(optarg,
+                                                       &params->ucp.send_datatype)) {
+            if (optarg2) {
```

Comment:
can move line 502 [ `optarg2 = strchr(optarg, ',')` ] in here, since it's relevant for UCP only anyway